ext/spl/tests/ArrayObject__serialize_saves_iterator_class.phpt

```diff
@@ -0,0 +1,15 @@
+--TEST--
+ArrayObject::__serialize saves any iterator class set by ::setIteratorClass
+--FILE--
+<?php
+
+class MyArrayIterator extends ArrayIterator {}
+$arrayObject = new ArrayObject(array(1, 2, 3));
+$arrayObject->setIteratorClass("MyArrayIterator");
+$serialized = serialize($arrayObject);
+$backAgain  = unserialize($serialized);
+echo $backAgain->getIteratorClass(), "\n";
```

Comment:
We should also check the additional error conditions in tests (see ext/spl/tests/unserialize_errors.phpt).

Reply:
Done.